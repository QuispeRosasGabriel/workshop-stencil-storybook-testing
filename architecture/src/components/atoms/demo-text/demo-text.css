:host {
  display: block;
  /* adding font-size */
  font-size: 18px;
}
