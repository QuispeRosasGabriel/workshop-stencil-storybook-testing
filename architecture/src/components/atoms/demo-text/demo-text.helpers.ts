import { text } from "@storybook/addon-knobs";

import { buildComponent, buildSpecPage } from "../../story.utils";

export const component = buildComponent("demo-text", {
  lastname: (prop = "source") => text("lastname", prop),
});

export const buildComponentTest = (
  newSpecPage,
  baseComponent,
  specParams = {}
) => buildSpecPage(baseComponent, newSpecPage, component, specParams);
