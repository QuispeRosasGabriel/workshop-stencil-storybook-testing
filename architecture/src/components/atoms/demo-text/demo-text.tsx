import { Component, Host, Prop, h, State } from "@stencil/core";

@Component({
  tag: "demo-text",
  styleUrl: "demo-text.css",
  shadow: true,
})
export class DemoText {
  @Prop() lastname: string;

  @State() isOpened: boolean;

  toggle() {
    this.isOpened = !this.isOpened;
  }

  render() {
    return <Host>{this.lastname}</Host>;
  }
}
