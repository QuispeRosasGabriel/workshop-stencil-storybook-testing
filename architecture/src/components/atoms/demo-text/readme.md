# demo-text



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type     | Default     |
| ---------- | ---------- | ----------- | -------- | ----------- |
| `lastname` | `lastname` |             | `string` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
