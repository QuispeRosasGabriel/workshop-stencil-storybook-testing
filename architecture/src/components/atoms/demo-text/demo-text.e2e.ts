import { E2EElement, newE2EPage } from "@stencil/core/testing";
import { DemoText } from "./demo-text";
import { buildComponentTest } from "./demo-text.helpers";

const component = buildComponentTest(newE2EPage, [DemoText]);

describe("should test e2e demo-text component", () => {
  let page: E2EElement;
  let element: E2EElement;

  describe("rendering", () => {
    it("should be able to render", async () => {
      page = await component({ lastname: "source" }, '');
      element = await page.find("demo-text");
      expect(element).toBeTruthy();
    });
  });

  describe("styles", () => {
    it("should match styles for component", async () => {
      page = await component({ lastname: "source" }, '');
      element = await page.find("demo-text");
      const styles = {
        fontSize: "18px",
      };
      expect(await element.getComputedStyle()).toMatchObject(styles);
    });
  });
});
