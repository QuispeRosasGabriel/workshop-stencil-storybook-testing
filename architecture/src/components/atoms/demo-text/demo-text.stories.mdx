import { withA11y } from "@storybook/addon-a11y";
import { Meta, Story } from "@storybook/addon-docs/blocks";
import { withKnobs } from "@storybook/addon-knobs";
import { withActions } from "@storybook/addon-actions";
import { withTests } from "@storybook/addon-jest";

import { component } from "./demo-text.helpers";
import readme from "./readme.md";

<Meta
  title="Atoms|Text"
  decorators={[withKnobs, withA11y]}
  parameters={{
    readme: { sidebar: readme },
    jest: ["demo-text.spec.ts"]
  }}
/>

# Atoms | Text

Documentacion acerca de demo-text

<Story name="Demo Text">{component({ lastname: "workshop" })}</Story>
