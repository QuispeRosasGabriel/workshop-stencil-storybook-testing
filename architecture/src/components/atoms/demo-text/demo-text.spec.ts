import { newSpecPage, SpecPage } from "@stencil/core/testing";

import { DemoText } from "./demo-text";
import { buildComponentTest } from "./demo-text.helpers";

const component = buildComponentTest(newSpecPage, [DemoText]);

describe("should test demo-text component", () => {
  describe("should render", () => {
    let page: SpecPage;

    it("should render by default with lastname: source", async () => {
      page = await component({ lastname: "source" }, '');
      expect(page.root.shadowRoot).toBeTruthy();
      expect(page.root).toEqualHtml(`
        <demo-text lastname="source">
          <mock:shadow-root>
            source
          </mock:shadow-root>
        </demo-text>
      `);
      expect(page.root).toMatchSnapshot();
    });

    it("should toggle isOpened", async () => {
      const page = new DemoText();
      expect(page.isOpened).toBeFalsy();
      page.toggle()
      expect(page.isOpened).toBeTruthy();
    });
  });
});
