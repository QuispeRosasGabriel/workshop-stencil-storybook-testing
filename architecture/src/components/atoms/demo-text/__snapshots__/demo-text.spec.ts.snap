// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test demo-text component should render should render by default with lastname: source 1`] = `
<demo-text lastname="source">
  <mock:shadow-root>
    source
  </mock:shadow-root>
</demo-text>
`;
